contracts/contracts/strategies/BaseCurveStrategy.sol

```diff
@@ -164,6 +140,58 @@ abstract contract BaseCurveStrategy is InitializableAbstractStrategy {
         IERC20(_asset).safeTransfer(_recipient, _amount);
     }
 
+    /**
+     * @dev Calculate amount of LP required when withdrawing specific amount of one
+     * of the underlying assets accounting for fees and slippage.
+     *
+     * Curve pools unfortunately do not contain a calculation function for
+     * amount of LP required when withdrawing a specific amount of one of the
+     * underlying tokens and also accounting for fees (Curve's calc_token_amount
+     * does account for slippage but not fees).
+     *
+     * Steps taken to calculate the metric:
+     *  - get amount of LP required if fees wouldn't apply
+     *  - increase the LP amount as if fees would apply to the entirety of the underlying
+     *    asset withdrawal. (when withdrawing only one coin fees apply only to amounts
+     *    of other assets pool would return in case of balanced removal - since those need
+     *    to be swapped for the single underlying asset being withdrawn)
+     *  - get amount of underlying asset withdrawn (this Curve function does consider slippage
+     *    and fees) when using the increased LP amount. As LP amount is slightly over-increased
+     *    so is amount of underlying assets returned.
+     *  - since we know exactly how much asset we require take the rate of LP required for asset
+     *    withdrawn to get the exact amount of LP.
+     */
+    function _calcCurveTokenAmount(uint256 _coinIndex, uint256 _amount)
+        internal
+        returns (uint256 required3Crv)
+    {
+        ICurvePool curvePool = ICurvePool(platformAddress);
+
+        uint256[3] memory _amounts = [uint256(0), uint256(0), uint256(0)];
+        _amounts[_coinIndex] = _amount;
+
+        // LP required when removing required asset ignoring fees
+        uint256 lpRequiredNoFees = curvePool.calc_token_amount(_amounts, false);
+        // LP required if fees would apply to entirety of removed amount
+        uint256 lpRequiredFullFees = lpRequiredNoFees.mulTruncateScale(
```

Comment:
Looking at [`StableMath` contract's code](https://github.com/mstable/mStable-contracts/blob/master/contracts/shared/StableMath.sol), it seems like `mulTruncateScale` requires three params, but only two are passed here?

Reply:
In solidity when "Using for" is used ([docs](https://docs.soliditylang.org/en/v0.8.17/contracts.html#using-for)) the docs say: 

The directive using A for B; can be used to attach functions (A) as member functions to any type (B). These functions will receive the object they are called on as their first parameter (like the self variable in Python).

Reply:
Ah yes, I'm sorry I missed that